doc/installation.md

```diff
@@ -0,0 +1,114 @@
+0. Place those two files side by side, because that's how WinSW discovers its configuration.
+0. Run `myapp.exe install <OPTIONS>` in order to install the service wrapper.
+0. Optional - Perform additional configuration in the Windows Service Manager.
+0. Optional - Perform extra configurations if required (guidelines are available below).
+ * Declare that the executable is compatible with .NET 4 or above
+ * Enable the WinSW offline mode
+0. Run the service from the Windows Service Manager.
+
+There are some details for each step available below.
+
+### Installation step details
+
+#### Step 2. Configuration file
+
+You write the configuration file that defines your service. 
+The example below is a primitive example being used in the Jenkins project:
+
+```
+    <service>
+      <id>jenkins</id>
+      <name>Jenkins</name>
+      <description>This service runs Jenkins continuous integration system.</description>
+      <env name="JENKINS_HOME" value="%BASE%"/>
+      <executable>java</executable>
+      <arguments>-Xrs -Xmx256m -jar "%BASE%\jenkins.war" --httpPort=8080</arguments>
+      <logmode>rotate</logmode>
+    </service>
+```
+
+The full specification of the configuration file is available [here](xmlConfigFile.md).
+
+#### Step 4. Service registration
+ 
+You can then install the service like:
+
+```
+    myapp.exe install <OPTIONS>
+```
+
+... and you can use the exit code from these processes to determine whether the operation was successful. 
+Possible return error codes are described  [here](http://msdn.microsoft.com/en-us/library/aa389390%28VS.85%29.aspx). 
+Beyond these error codes, all the non-zero exit code should be assumed as a failure.
+
+The Installer can be also started with the `/p` option.
+In such case it will prompt for an account name and password, which should be used as a service account.
+
+#### Step 5. Windows Service Manager
+
+Once the service is installed, you can start it from Windows Service Manager.
+If you open `Properties` for the service, you can also configure how the service should be launched. 
+
+In particular, the following option can be set up:
+
+* Service automatic startup on the Windows startup
+* User or system account, under which the service runs
+* Recovery options (how Windows recovers the service if it dies due to whatever reason)
+
+In addition to the service manager, it is possible to make some additional configurations in the `Windows Registry Editor`.
+
+Once the start button is clicked, Windows will start `myapp.exe`, 
+  then `myapp.exe` will launch the executable specified in the configuration file (Java in this case). 
+  If this process dies, `myapp.exe` will exit itself, and the service will be considered stopped.
+  
+### Extra configuration options
+  
+#### Making WinSW compatible with .NET runtime 4.0+
+
+Modern versions of Windows (e.g. Windows Server 2012 or Windows 10) do not ship with .NET runtime `2.0`, which is what `winsw.exe` is built against. 
```

Comment:
I thought this was the default?  If not why don't we just embed this info in the binary?